src/main/scala/magellan/Polygon.scala

```diff
@@ -128,7 +128,14 @@ class Polygon extends Shape {
     * @return
     */
   private [magellan] def intersects(line: Line): Boolean = {
-    loops.exists(_.intersects(line))
+    var intersects = false
+    if(this.contains(line.getStart()) || this.contains(line.getEnd())){
```

Comment:
ok, i'm worried about how intersects is strictly defined: i.e., if a line is contained fully within the polygon, within = true, but is intersection also = true?
it would be great to add a test using ESRI Java API (following similar tests in PolygonSuite) to exhibit how ESRI interprets line intersects polygon